app/schema.graphql

```diff
@@ -0,0 +1,103 @@
+scalar Datetime
+
+type Author {
+  name: String
+  email: String
+}
+
+# git commit data.
+type Commit {
+  sha1: String!
+  author: Author!
+  date: Datetime!
+  message: String!
+  refs: [Ref]
+  objectName: ObjectName!
+}
+
+# file info and metadata.
+type Info {
+  # base name of the file.
+  name: String!
+  # length in bytes.
+  size: Int!
+  # if it's a directory.
+  isdirectory: Boolean!
+  # any flags git stores, currently only executable.
+  flags: [Int]!
+  # last commit which affected this path.
+  lastcommit: Commit!
+}
+
+type BlobEntry {
+  content: String!
+  info: Info!
+}
+
+type TreeEntry {
+  path: String!
+  info: Info!
+}
+
+# Representation of a refname. Special refs like HEAD are omitted on purpose as
+# of now.
+#
+# master
+# v1.0.0
+# heads/master
+# tags/v1.0.0
+# remotes/origin/master
+type Ref = {
+  name: String!
+  sha1: String!
+}
+
+type Query {
+  # Source browsing.
+  blob(projectId: ID!, revision: String!, path: String!): BlobEntry
+  tree(projectId: ID!, revision: String!, prefix: String!): [TreeEntry!]!
+
+  # Commits browsing
+  commits(projectId: ID!, Revision: String!): [Commit]!
+  commit(projectId: ID!, sha1: String!) Commit
+
+  # Ref browsing
+  branches(projectId: ID!): [Ref]!
+  refs(projectId: ID!): [Ref]!
+  tags(projectId: ID!): [Ref]!
+}
+
+################
+# Exploration. #
+
+enum ObjectType {
+  Commit
+  Tree
+  Blob
+  Tag
+}
+
+# git tags.
+type LightweightTag {
```

Comment:
Again, need to flesh out how to present these to you via `radicle-surf` :thought_balloon:

Reply:
Not at all urgent, this is still from times when I went down the git object types to extract their shapes. We don't currently have any tag objects surfaced, besides the refnames. It will become more interesting down the road and esp. the power of annotated tags and the ability to stick tree objects in there.